virtool/dispatcher/connection.py

```diff
@@ -30,8 +30,9 @@ async def send(self, message: Union[dict, list]):
 
             await self.close()
 
-    async def close(self):
+    async def close(self, code: Union[int, str] = None):
```

Comment:
1. Should we just required `int` codes? I don't see the point of allowing both types here.
2. Should this be a keyword argument? Are sometimes not passing `code` to `Connection.close()`?

Reply:
1. Yeah seems fair, limited to `int` in next push
2. I gave a default value since we aren't always passing a code, although perhaps we should be. The other instance where the server will proactively close the connection can probably be covered by the code for the server going down (1001).  I will add the code and remove the default value.